charts/spire/charts/spire-server/templates/configmap.yaml

```
@@ -46,11 +47,41 @@ plugins:
               service_account_allow_list: {{ include "spire-server.serviceAccountAllowedList" $root | trim }}
   {{- end }}
   {{- end }}
-
+ 
+  {{- with .Values.keyManager.disk }}
+  {{- if eq (.enabled | toString) "true" }}
+  {{- $keyManagerUsed = add1 $keyManagerUsed }}
   KeyManager:
     - disk:
         plugin_data:
-          keys_path: "/run/spire/data/keys.json"
+          keys_path: {{ .keysPath | quote }}
+  {{- end }}
+  {{- end }}
+
+  {{- with .Values.keyManager.awsKMS }}
+  {{- if eq (.enabled | toString) "true" }}
+  {{- $keyManagerUsed = add1 $keyManagerUsed }}
+  KeyManager:
+    - aws_kms:
+        plugin_data:
+          region: {{ .region | quote }}
+          key_metadata_file: {{ .keyMetadataFile | quote }}
+          {{- if ne .accessKeyID "" }}
+          access_key_id: {{ .accessKeyID | quote }}
```

This is placing the potentially senstive key/secret into the configmap. They should be in a secret instead. Here, you could do:
```suggestion
          access_key_id: "${AWSKM_ACCESS_KEY_ID}"
```